src/Umbraco.Cms.Integrations.SEO.SemrushTools/Umbraco.Cms.Integrations.SEO.SemrushTools/App_Plugins/UmbracoCms.Integrations/SEO/SemrushTools/semrushToolsController.js

```diff
@@ -0,0 +1,250 @@
+                vm.showError("Search Keywords", "Please select a data source and method.");
+                return;
+            }
+
+            searchRelatedPhrases(1);
+        }
+
+        // authorization listener
+        window.addEventListener("message", function (event) {
+            if (event.data.type === "semrush:oauth:success") {
+
+                var codeParam = "?code=";
+
+                vm.showSuccess("SEMrush API", "Access Approved");
+
+                vm.authWindow.close();
+
+                var code = event.data.url.slice(event.data.url.indexOf(codeParam) + codeParam.length);
+
+                umbracoCmsIntegrationsSemrushResources.getAccessToken(code).then(function (response) {
+                    if (response !== "error") vm.isConnected = true;
+                });
+
+
+            } else if (event.data.type === "semrush:oauth:denied") {
+                vm.showError("SEMrush API", "Access Denied");
+
+                vm.authWindow.close();
+
+                revokeToken();
+            }
+
+        }, false);
+
+        function revokeToken() {
+            umbracoCmsIntegrationsSemrushResources.revokeToken().then(function () {
+                vm.isConnected = false;
+            });
+        }
+
+        function searchRelatedPhrases(pageNumber) {
+            vm.loading = true;
+            umbracoCmsIntegrationsSemrushResources.getRelatedPhrases(vm.SelectedPropertyDetails.value, pageNumber, vm.SelectedDataSource, vm.SelectedMethod)
+                .then(function (response) {
+
+                    vm.loading = false;
+
+                    if (response.IsSuccessful === false) {
+                        vm.showError("SEMrush API", response.error);
+                    } else {
+                        vm.pagination = {
+                            pageNumber: vm.pagination.pageNumber,
+                            totalPages: response.TotalPages
+                        };
+
+                        vm.relatedPhrasesList = response;
+                    }
+                });
+        }
+
+        // notifications
+        vm.showSuccess = function (headline, message) {
+            notificationsService.success(headline, message);
+        }
+
+        vm.showWarning = function (headline, message) {
+            notificationsService.warning(headline, message);
+        }
+
+        vm.showError = function (headline, message) {
+            notificationsService.error(headline, message);
+        }
+
+        // pagination
+        function nextPage(pageNumber) {
+            searchRelatedPhrases(pageNumber);
+        }
+
+        function prevPage(pageNumber) {
+            searchRelatedPhrases(pageNumber);
+        }
+
+        function changePage(pageNumber) {
+            searchRelatedPhrases(pageNumber);
+        }
+
+        function goToPage(pageNumber) {
+            searchRelatedPhrases(pageNumber);
+        }
+
+        // status
+        vm.OnViewStatus = function () {
```

Comment:
Just for consistency, think these methods should be camel-cased, e.g. `onViewStatus`.

Reply:
Item addressed.